### PR TITLE
Do not handle input in empty ScrollContainer

### PR DESCRIPTION
**Pull request by UselessToucan**
resolves https://github.com/ppy/osu-framework/issues/3245

### CODE REVIEW

File: osu.Framework/Graphics/Containers/Container.cs [hunk 1]
```diff
@@ -50,6 +51,12 @@ public Container()
                 aliveInternalChildrenAsT = new LazyList<Drawable, T>(AliveInternalChildren, c => (T)c);
         }
 
+        internal override bool BuildNonPositionalInputQueue(List<Drawable> queue, bool allowBlocking = true) =>
+            AliveInternalChildren.Any() && base.BuildNonPositionalInputQueue(queue, allowBlocking);
+
+        internal override bool BuildPositionalInputQueue(Vector2 screenSpacePos, List<Drawable> queue) =>
+            AliveInternalChildren.Any() && base.BuildPositionalInputQueue(screenSpacePos, queue);
```
Comment (iiSaLMaN) [inline osu.Framework/Graphics/Containers/Container.cs]: I think you're looking for overriding `PropagateNonPositionalInputSubTree` and `PropagatePositionalInputSubTree` instead.

Comment (UselessToucan) [reply osu.Framework/Graphics/Containers/Container.cs]: Never used these methods. Will check them out tomorrow.

**Comment by UselessToucan** [comment]
>     * Why not on `CompositeDrawable` instead?

Because the issue is about `ScrollContainer`. I decided that `CompositeDrawable` level is too wide for this case.

>     * This should probably have tests to validate this solution and not regress on the issue further on.

Indeed. I will add them tomorrow.

**Comment by smoogipoo** [comment]
I disagree that this is what we want. It’s possible that we want composites to handle input when there’s no alive children. Something like moving around the screen (e.g. maybe a “blueprint” that overlays the drawable but needs to know about the drawable hover state, or the drawable itself is handling this), or perhaps adding a child on mouse/key events.

The issue was specific to ScrollContainer, not intended as a general issue.

**Comment by peppy** [comment]
unsure if it should be done at the input queue building, as opposed to the event overrides, where we would usually do this kind of thing.

**Pull request merged**